modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/unit/FeatureFlagBatchEngineUnitProcessor.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.batch.engine.internal.unit;
+
+import com.liferay.petra.executor.PortalExecutorManager;
+import com.liferay.petra.function.UnsafeSupplier;
+import com.liferay.portal.kernel.feature.flag.FeatureFlag;
+import com.liferay.portal.kernel.feature.flag.listener.FeatureFlagListener;
+import com.liferay.portal.kernel.util.Tuple;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carlos Correa
+ */
+@Component(service = FeatureFlagBatchEngineUnitProcessor.class)
+public class FeatureFlagBatchEngineUnitProcessor {
+
+	public void registerBatchEngineUnit(
+		long companyId, String featureFlagKey,
+		UnsafeSupplier<CompletableFuture<Void>, Exception> unsafeSupplier) {
+
+		_unsafeSuppliers.compute(
```

Comment:
This can be simplified:

```java
List<UnsafeSupplier<CompletableFuture<Void>, Exception>>
		unsafeSuppliers = _unsafeSuppliers.computeIfAbsent(
			_getTuple(companyId, featureFlagKey), (key) -> new ArrayList<>());
		
unsafeSuppliers.add(unsafeSupplier);
```